results/pkg/api/server/v1alpha1/server_test.go

```diff
@@ -72,6 +72,16 @@ func TestGetResult(t *testing.T) {
 }
 
 func TestUpdateResult(t *testing.T) {
+	t.Run("Test etag generation", func(t *testing.T) {
+		r1 := Result(&ppb.TaskRun{})
+		etag1 := obtainResultEtag(t, r1)
+		r1.Annotations = map[string]string{"foo": "bar"}
+		etag2 := obtainResultEtag(t, r1)
+		if etag1 == etag2 {
+			t.Fatalf("different results can't have the same etag")
```

Comment:
We probably want a more exact test here that compares the expected etag output. Technically this could still pass as long as the etag generation was non-deterministic. 2 test cases come to mind:

1) Result with no etag, verify this is filled in and set properly.
2) Result with etag, verify this is overridden.